channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -670,29 +683,65 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControls {
                 try {
                     let onCompleteFuncs = {
                         xdrOnComplete: (xdr: IXDomainRequest, oncomplete: OnCompleteCallback,payload?: IPayloadData) => {
-                            let data = _getPayloadArr(payload);
-                            if (!data) {
+                            let payloadArr = _getPayloadArr(payload);
+                            if (!payloadArr) {
                                 return;
                             }
-                            return _xdrOnLoad(xdr, data);
+                            const responseText = _getResponseText(xdr);
+                            let statsbeat = _core.getStatsBeat();
+                            var endpointHost = urlParseUrl(_self._senderConfig.endpointUrl).hostname;
+                            if (xdr && (responseText + "" === "200" || responseText === "")) {
+                                _consecutiveErrors = 0;
+                                if (statsbeat) {
+                                    statsbeat.count(200, payload, endpointHost);
+                                }
+                            } else {
+                                const results = parseResponse(responseText);
+                    
+                                if (results && results.itemsReceived && results.itemsReceived > results.itemsAccepted
+                                    && !_isRetryDisabled) {
+                                    if (statsbeat) {
+                                        statsbeat.count(201, payload, endpointHost);
+                                    }
```

Comment:
For 206 (partial success) we will also be adding a design that will parse the actual failed/succeeded items in the batch. I believe the logic for receiving 206 is:

1. Increment success by 1 for success
2. Increment retry by 1 if there are ANY retryable envelopes in rejected items
3. Increment failure by 1 if there are ANY failed (non-retryable) envelopes in rejected items
4. Increment throttle by 1 if there are ANY throttled response codes in rejected items.

@JacksonWeber  please correct me if I missed anything above.

The logic above has not been added to the specs yet so I don't believe we need to include this for the first iteration but it would be good to create a tracking issue for it @siyuniu-ms @MSNev .